### PR TITLE
update rpk generate grafana dashboards usage for 23.2

### LINKED ISSUE #1394: rpk: generate grafana-dashboards improvements and Grafana Labs
### Describe the Issue

Currently, our docs (https://docs.redpanda.com/docs/reference/internal-metrics/#configure-grafana-internal-metrics) point to generating the Grafana Dashboard using the command `rpk generate grafana-dashboard`, however, we are now uploading a set of improved dashboards to Grafana Labs in https://grafana.com/grafana/dashboards/?search=redpanda

We have a set of issues to change the way things work in rpk:
- https://github.com/redpanda-data/redpanda/issues/6502
- https://github.com/redpanda-data/redpanda/issues/9485

The goal of this ticket is to:

1. Change the documentation to point to the Grafana labs page.
2. Document the new features in rpk once the two tickets listed above are implemented.

SME: @r-vasquez

### PR DESCRIPTION
**Pull request by r-vasquez**
Fixes #1394

### CODE REVIEW

File: versioned_docs/version-23.2/manage/monitoring.mdx [hunk 1]
```diff
@@ -29,11 +29,34 @@ When monitoring internal metrics, consider enabling [aggregate_metrics](../../re
...
+    - [Leadership changes](#leadership-changes)
+    - [Under-replicated partitions](#under-replicated-partitions)
+    - [Leaderless partitions](#leaderless-partitions)
+    - [Raft RPCs](#raft-rpcs)
+  - [Consumers](#consumers)
+    - [Consumer group lag](#consumer-group-lag)
+  - [Services](#services)
+    - [Schema registry](#schema-registry)
+    - [REST proxy](#rest-proxy)
+- [References](#references)
```
Comment (shorgi) [inline versioned_docs/version-23.2/manage/monitoring.mdx:59]: @r-vasquez - I'm guessing you're using an IDE like VSCode that auto-formats markdown to include the entire Table of Contents inline? If you are, and these added headings are unintentional, need to clean these up (and perhaps [disable the auto formatting in VSCode](https://whatismarkdown.com/how-to-disable-autoformatting-in-visual-studio-code-for-markdown/#:~:text=Find%20the%20%E2%80%9CMarkdown%3A%20Enable%20Auto,documents%20when%20you%20save%20them))

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 2]
```diff
@@ -5,9 +5,83 @@ title: rpk generate grafana-dashboard
 <head>
   <meta name="title" content="rpk generate grafana-dashboard | Redpanda Docs" />
   <meta name="description" content="rpk generate grafana-dashboard" />
+  <link rel="canonical" href="https://docs.redpanda.com/docs/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard/" />
 </head>
 
-Generate a Grafana dashboard for Redpanda metrics.
+Generate Grafana Dashboards for Redpanda Metrics
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:11]:
```suggestion
Generate Grafana dashboards for Redpanda metrics.
```

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 3]
```diff
@@ -5,9 +5,83 @@ title: rpk generate grafana-dashboard
...
+
+Use this command to generate sample Grafana dashboards for Redpanda metrics. 
+These dashboards can be imported into a Grafana or Grafana Cloud instance.
+
+To select a specific dashboard, use the '--dashboard' flag followed by the
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:16]:
```suggestion
To select a specific dashboard, use the `--dashboard` flag followed by the 
```

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 4]
```diff
@@ -5,9 +5,83 @@ title: rpk generate grafana-dashboard
...
+To select a specific dashboard, use the '--dashboard' flag followed by the 
+dashboard name. For example, to generate the operations dashboard, run:
+
+```
+rpk generate grafana-dashboard --dashboard operations
+```
+
+The selected dashboard will be downloaded from our GitHub repository:
+
+https://github.com/redpanda-data/observability
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:25]:
```suggestion
The selected dashboard will be downloaded from Redpanda Data's [observability GitHub repository](https://github.com/redpanda-data/observability).
```

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 5]
```diff
@@ -5,9 +5,83 @@ title: rpk generate grafana-dashboard
...
+
+Note that the legacy dashboard is still available as an option, and will not be 
+downloaded from github. Instead, the dashboard will be generated based on the 
+metrics endpoint used.
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:29]:
```suggestion
:::note
The legacy dashboard is still available as an option (`legacy`), but it isn't downloaded from GitHub. Instead, the generated dashboard is based on which metrics endpoint is used (`--metrics-endpoint`).
:::
```

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 6]
```diff
@@ -5,9 +5,83 @@ title: rpk generate grafana-dashboard
...
+
+### Available Dashboards
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:31]:
```suggestion
### Available dashboards
```

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 7]
```diff
@@ -5,9 +5,83 @@ title: rpk generate grafana-dashboard
...
+
+You can select one of the following dashboard types:
+
+<table>
+  <tbody>
+    <tr>
+      <td>
+        <strong> Name</strong>
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:39]:
```suggestion
        <strong>Name</strong>
```

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 8]
```diff
@@ -5,9 +5,83 @@ title: rpk generate grafana-dashboard
...
+      </td>
+      <td>
+        <strong> Description</strong>
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:42]:
```suggestion
        <strong>Description</strong>
```

File: versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx [hunk 9]
```diff
@@ -31,11 +105,10 @@ rpk generate grafana-dashboard [flags]
       </td>
     </tr>
     <tr>
-      <td>--datasource</td>
+      <td>--dashboard</td>
       <td>string</td>
       <td>
-        The name of the Prometheus datasource as configured in your Grafana
-        instance.
+        The name of the dashboard you wish to download; use --dashboard help for more info (default "operations")
```
Comment (shorgi) [inline versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx:111]:
```suggestion
        The name of the dashboard you wish to download. Use --dashboard help for more info (default "operations")
```

**Review by shorgi — APPROVED** [review]
Thanks @r-vasquez for this PR. See my comments about minor edits.

**Comment by r-vasquez** [comment]
@shorgi I committed the suggestions, should I squash the commits? or do we have auto squash on merge?

**Comment by shorgi** [comment]
> @shorgi I committed the suggestions, should I squash the commits? or do we have auto squash on merge?

Thanks @r-vasquez for checking, we `Squash and merge` docs PRs, I'll go ahead and do it for you.

**Pull request merged**